sdk/servicebus/service-bus/test/utils/aadUtils.ts

```diff
@@ -1,111 +1,64 @@
-import msRestNodeAuth from "@azure/ms-rest-nodeauth";
-import { ServiceBusManagementClient } from "@azure/arm-servicebus";
-import { SBQueue, SBTopic, SBSubscription } from "@azure/arm-servicebus/esm/models";
-import { EnvVarKeys, getEnvVars } from "./envVarUtils";
-
-const _loginWithServicePrincipalSecret = msRestNodeAuth.loginWithServicePrincipalSecret;
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
 
-export const loginWithServicePrincipalSecret = _loginWithServicePrincipalSecret;
+import { ServiceBusAtomManagementClient } from "../../src/serviceBusAtomManagementClient";
+import { EnvVarKeys, getEnvVars } from "./envVarUtils";
+import { QueueOptions, TopicOptions, SubscriptionOptions } from "../../src";
 
-export async function recreateQueue(queueName: string, parameters: SBQueue): Promise<void> {
+export async function recreateQueue(queueName: string, parameters: QueueOptions): Promise<void> {
   const env = getEnvVars();
-  await msRestNodeAuth
-    .loginWithServicePrincipalSecret(
-      env[EnvVarKeys.AAD_CLIENT_ID],
-      env[EnvVarKeys.AAD_CLIENT_SECRET],
-      env[EnvVarKeys.AAD_TENANT_ID]
-    )
-    .then(async (creds) => {
-      const client = await new ServiceBusManagementClient(
-        creds,
-        env[EnvVarKeys.AZURE_SUBSCRIPTION_ID]
-      );
-      await client.queues.deleteMethod(
-        env[EnvVarKeys.RESOURCE_GROUP],
-        getNamespace(env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]),
-        queueName,
-        function(error: any): void {
-          if (error) throw error.message;
-        }
-      );
-      await client.queues.createOrUpdate(
-        env[EnvVarKeys.RESOURCE_GROUP],
-        getNamespace(env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]),
-        queueName,
-        parameters,
-        function(error: any): void {
-          if (error) throw error.message;
-        }
-      );
-    });
+  if (env[EnvVarKeys.CLEAN_NAMESPACE]) {
+    const client = await new ServiceBusAtomManagementClient(
+      env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]
+    );
+
+    try {
+      await client.deleteQueue(queueName);
+    } catch (err) {
+      // Delete and ignore if already deleted
```

Comment:
If the delete failed due to any reason other than the entity not existing, then we need those to surface. Isnt there a status code that we can look for and ignore only that?

Reply:
If the `before` / `after` hooks fail, the tests do not yield any meaningful errors since multiple create/delete operations are taking place in these hooks currently.
I think we can have this optimization once we refactor tests to self contain the before each after each activities.

Reply:
The other concern is that if for any reason there's been some network issues and some entities get left over, it would require manual intervention else the tests will keep failing.
Since we already have test coverage for the test resources creation/deletion - wouldn't it be more resilient to ensure the test resources are cleaned up each time (by creating if/if not existing and eventually delete if/if not existing..). So even if delete failed in one run, it would eventually get deleted in subsequent runs. cc: @mikeharder @KarishmaGhiya any thoughts/recommendation?

Reply:
Eventually, we want to move to a state where each test uses entities with a unique name and so the same entity will not be re-used between tests. Therefore, it would be the responsibility of each tests's `after` hook to clean up the entities created in the `before` hook

In this PR, my concern was mainly around delete errors being ignored causing tests to give unexpected results. But, since there is a create step which will fail if the delete step didn't succeed, we are good for now, for this PR.